components/ws-proxy/pkg/proxy/routes.go

```diff
@@ -106,25 +112,61 @@ func installTheiaRoutes(r *mux.Router, config *RouteHandlerConfig, rh *RouteHand
 	rh.theiaFileHandler(r.PathPrefix("/files").Subrouter(), config)
 
 	rh.theiaHostedPluginHandler(r.PathPrefix("/hostedPlugin").Subrouter(), config)
+	rh.theiaWebviewHandler(r.PathPrefix("/webview").Subrouter(), config)
 
 	rh.supervisorUnauthenticatedAPIHandler(r.PathPrefix("/_supervisor/v1/status/supervisor").Subrouter(), config)
 	rh.supervisorUnauthenticatedAPIHandler(r.PathPrefix("/_supervisor/v1/status/ide").Subrouter(), config)
 	rh.supervisorAuthenticatedAPIHandler(r.PathPrefix("/_supervisor/v1").Subrouter(), config)
 	rh.supervisorAuthenticatedAPIHandler(r.PathPrefix("/_supervisor").Subrouter(), config)
 
-	rh.theiaWebviewHandler(r.PathPrefix("/webview").Subrouter(), config)
-
+	// TODO(cw): we just enable the IDE host route if blobserver is active. Once blobserve is standard,
+	//           remove this branch and always register the handler.
+	if config.Config.BlobServer != nil {
+		rh.supervisorIDEHostHandler(r.Path("/").MatcherFunc(matchIDEQuery(false)).Subrouter(), config)
+		rh.supervisorIDEHostHandler(r.Path("/index.html").MatcherFunc(matchIDEQuery(false)).Subrouter(), config)
+	}
 	rh.theiaRootHandler(r.NewRoute().Subrouter(), config)
 }
 
+func matchIDEQuery(mustBePresent bool) mux.MatcherFunc {
+	return func(req *http.Request, match *mux.RouteMatch) bool {
+		_, present := req.URL.Query()[ideIndexQueryMarker]
```

Comment:
Does it also drop `gitpod-ide-index` from query when it is present? The thing is that query can be just JSON payload generated by another IDE window. If we don't clean up the query IDE won't be able to parse the payload.

Reply:
I'm dropping it in the request towards blobserve - the only place I can do so. 
However, I think that's pretty inconsequential as that query will only be seen by blobserve, which isn't really bothered.